File: code/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/services/lookup/CollapseStyleLookupCall.java
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.scout.widgets.client.services.lookup;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import org.eclipse.scout.rt.client.ui.group.IGroup;
import org.eclipse.scout.rt.platform.classid.ClassId;
import org.eclipse.scout.rt.shared.services.lookup.ILookupRow;
import org.eclipse.scout.rt.shared.services.lookup.LocalLookupCall;
import org.eclipse.scout.rt.shared.services.lookup.LookupRow;

@ClassId("988bc56a-343a-4842-a0b2-110734072c95")
public class CollapseStyleLookupCall extends LocalLookupCall<String> {

  private static final long serialVersionUID = 1L;

  @Override
  protected List<? extends ILookupRow<String>> execCreateLookupRows() {
    return Stream
        .of(IGroup.COLLAPSE_STYLE_LEFT, IGroup.COLLAPSE_STYLE_RIGHT, IGroup.COLLAPSE_STYLE_BOTTOM)
        .map(s -> new LookupRow<>(s, s))
        .collect(Collectors.toList());
  }
}

File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/accordion/CollapseStyleLookupCall.ts
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
import {Group, GroupCollapseStyle, StaticLookupCall} from '@eclipse-scout/core';

export class CollapseStyleLookupCall extends StaticLookupCall<GroupCollapseStyle> {

  constructor() {
    super();
  }

  protected override _data(): any[] {
    return CollapseStyleLookupCall.DATA;
  }

  static DATA = [
    [Group.CollapseStyle.LEFT, 'left'],
    [Group.CollapseStyle.RIGHT, 'right'],
    [Group.CollapseStyle.BOTTOM, 'bottom']
  ];
}
